components/win-users/src/sid.rs

```diff
@@ -0,0 +1,335 @@
+    aclBytesInUse: DWORD,
+    aclBytesFree:  DWORD,
+}
+
+#[repr(C)]
+struct ACE_HEADER {
+    aceType:  BYTE,
+    aceFlags: BYTE,
+    aceSize:  WORD,
+}
+type PaceHeader = *mut ACE_HEADER;
+
+#[repr(C)]
+struct ACCESS_ALLOWED_ACE {
+    header:   ACE_HEADER,
+    mask:     ACCESS_MASK,
+    sidStart: DWORD,
+}
+
+extern "system" {
+    fn ObtainSid(hToken: HANDLE, psid: *mut PSID) -> BOOL;
+}
+
+#[link(name = "user32")]
+extern "system" {
+    fn AddAccessAllowedAceEx(pAcl: PACL,
+                             dwAceRevision: DWORD,
+                             aceFlags: DWORD,
+                             accessMask: DWORD,
+                             pSid: PSID)
+                             -> BOOL;
+    fn AddAce(pAcl: PACL,
+              dwAceRevision: DWORD,
+              dwStartingAceIndex: DWORD,
+              pAceList: LPVOID,
+              nAceListLength: DWORD)
+              -> BOOL;
+    fn ConvertSidToStringSidW(Sid: PSID, StringSid: LPCWSTR) -> BOOL;
+    fn GetAce(pAcl: PACL, dwAceIndex: DWORD, pAce: *mut LPVOID) -> BOOL;
+    fn GetAclInformation(pAcl: PACL,
+                         pAclInformation: LPVOID,
+                         nAclInformationLength: DWORD,
+                         dwAclInformationClass: DWORD)
+                         -> BOOL;
+    fn OpenProcessToken(processHandle: HANDLE, desiredAccess: DWORD, tokenHandle: PHANDLE) -> BOOL;
+    fn GetLengthSid(pSid: PSID) -> DWORD;
+    fn GetSecurityDescriptorDacl(pSecurityDescriptor: PSECURITY_DESCRIPTOR,
+                                 lpbDaclPresent: LPBOOL,
+                                 pDacl: *mut PACL,
+                                 lpbDaclDefaulted: LPBOOL)
+                                 -> BOOL;
+    fn GetUserObjectSecurity(hObj: HANDLE,
+                             pSIRequested: PSECURITY_INFORMATION,
+                             pSD: PSECURITY_INFORMATION,
+                             nLength: DWORD,
+                             lpnLengthNeeded: LPDWORD)
+                             -> BOOL;
+    fn InitializeAcl(pAcl: PACL, nAclLength: DWORD, dwAclRevision: DWORD) -> BOOL;
+    fn InitializeSecurityDescriptor(pSecurityDescriptor: PSECURITY_DESCRIPTOR,
+                                    dwRevision: DWORD)
+                                    -> BOOL;
+    fn SetSecurityDescriptorDacl(pSecurityDescriptor: PSECURITY_DESCRIPTOR,
+                                 bDaclPresent: BOOL,
+                                 pDacl: PACL,
+                                 bDaclDefaulted: BOOL)
+                                 -> BOOL;
+    fn SetUserObjectSecurity(hObj: HANDLE,
+                             pSIRequested: PSECURITY_INFORMATION,
+                             pSID: PSECURITY_DESCRIPTOR)
+                             -> BOOL;
+}
+
+pub const GENERIC_READ: DWORD = 0x8000_0000;
+pub const GENERIC_WRITE: DWORD = 0x4000_0000;
+pub const GENERIC_EXECUTE: DWORD = 0x2000_0000;
+pub const GENERIC_ALL: DWORD = 0x1000_0000;
```

Comment:
I'm probably missing something, but converting them to a binary representation doesn't make it clearer that they're correct to me.  If anything, it feels more visually cluttered to my eyes.  I'm hesitant to make any more changes to this code than we need to to get clippy passing.  This might be something worth following up on at a later date.

Reply:
This may just be personal preference, and it definitely gets harder with larger values, but when trying to set up bit flags, compare:

```rust
        const FOO = 0b0001;
        const BAR = 0b0010;
        const BAZ = 0b0100;
        const QUX = 0b1000;
```
```rust
        const FOO = 0x1;
        const BAR = 0x2;
        const BAZ = 0x4;
        const QUX = 0x8;
```
I just like the way the former allows me to visually scan down the columns and see that they're correct rather than doing math in my head. In the simple case, maybe it's not as valuable, but with compound values like `0x37F` it becomes much nicer. I have no idea off the top of my head what `0x37F` looks like in binary or what flags it might turn on, but the literal binary representation makes that much friendlier for me.

Reply:
Thank you for the example.  I understand your point much better now 😁.  These just read like magic numbers to me.